### PR TITLE
ROX-18155: pg generic store: Delete

### PR DESCRIPTION
**Pull request by janisz**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6766** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6766" target="_blank">Graphite</a> 
    * **PR #6767** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6767" target="_blank">Graphite</a> 
      * **PR #6768** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6768" target="_blank">Graphite</a> 
        * **PR #6769** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6769" target="_blank">Graphite</a> 
          * **PR #6770** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6770" target="_blank">Graphite</a> 
            * **PR #6771** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6771" target="_blank">Graphite</a> 
              * **PR #6772** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6772" target="_blank">Graphite</a> 
                * **PR #6773** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6773" target="_blank">Graphite</a> 
                  * **PR #6774** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6774" target="_blank">Graphite</a> 
                    * **PR #6775** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6775" target="_blank">Graphite</a>  👈
                      * **PR #6776** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6776" target="_blank">Graphite</a> 
                        * **PR #6777** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6777" target="_blank">Graphite</a> 
                          * **PR #6778** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6778" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6775?utm_source=stack-comment).

File: pkg/search/postgres/store.go [hunk 1]
```diff
@@ -377,3 +377,9 @@ func (s *GenericStore[T, PT]) DeleteByQuery(ctx context.Context, query *v1.Query
 
 	return RunDeleteRequestForSchema(ctx, s.schema, q, s.db)
 }
+
+// Delete removes the object associated to the specified ID from the store.
+func (s *GenericStore[T, PT]) Delete(ctx context.Context, id string) error {
+	q := search.NewQueryBuilder().AddDocIDs(id).ProtoQuery()
+	return s.DeleteByQuery(ctx, q)
+}
```
Comment (rhybrillou) [inline pkg/search/postgres/store.go:385]: Nice simplification and code reduction :o)

**Review by rhybrillou — APPROVED** [review]

File: pkg/search/postgres/store.go (hunk 1 above)
Comment (janisz) [reply pkg/search/postgres/store.go:375]: On the other hand it prevents us from tracking simple queries from complicated ones

**Comment by janisz** [comment]
[Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6775) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Comment by janisz** [comment]
@janisz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6775).

**Pull request merged**